Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogueBehavior.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using Discord;
+using Discord.WebSocket;
+using Microsoft.Extensions.Caching.Memory;
+using Modix.Common.Messaging;
+using Modix.Data.Models.Core;
+using Modix.Data.Models.Promotions;
+using Modix.Services.Core;
+using Modix.Services.Images;
+using Modix.Services.Promotions;
+using Modix.Services.Utilities;
+
+namespace Modix.Bot.Behaviors
+{
+    public class PromotionDialogueBehavior :
+        INotificationHandler<ReactionAddedNotification>,
+        INotificationHandler<PromotionActionCreatedNotification>
+    {
+        private readonly DateTime _utcNow = DateTime.UtcNow;
+
+        private static readonly Emoji _approveEmoji = new Emoji("👍");
+
+        private static readonly Emoji _disproveEmoji = new Emoji("👎");
+
+        private IDesignatedChannelService _designatedChannelService { get; }
+
+        private IDiscordSocketClient _discordSocketClient { get; }
+
+        private IMemoryCache _memoryCache { get; }
+
+        private IPromotionsService _promotionsService { get; }
+
+        private IImageService _imageService { get; }
+
+        public PromotionDialogueBehavior(IDesignatedChannelService designatedChannelService,
+            IDiscordSocketClient discordSocketClient,
+            IMemoryCache memoryCache,
+            IPromotionsService promotionsService,
+            IImageService imageService)
+
+        {
+            _designatedChannelService = designatedChannelService;
+            _discordSocketClient = discordSocketClient;
+            _memoryCache = memoryCache;
+            _promotionsService = promotionsService;
+            _imageService = imageService;
+        }
+
+        public async Task HandleNotificationAsync(PromotionActionCreatedNotification notification,
+            CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return;
+
+            switch (notification.Data.Type)
+            {
+                case PromotionActionType.CampaignCreated:
+                    await CreatePromoPoll(notification.Data);
+                    break;
+                case PromotionActionType.CampaignClosed:
+                    await DeletePromoPoll(notification.Data);
+                    break;
+            }
+        }
+
+        public async Task HandleNotificationAsync(ReactionAddedNotification notification,
+            CancellationToken cancellationToken = default)
+        {
+            var reaction = notification.Reaction.Emote;
+            var user = (IGuildUser)notification.Reaction.User.Value;
```

Comment:
This is dangerous on 2 levels. A) The user might not be there, and you get an exception calling `.Value`. B) The user might not be an `IGuildUser` and you get a cast exception. Pull these checks into the bail-out logic below.